dags/veda_data_pipeline/groups/discover_group.py

```diff
@@ -42,60 +43,59 @@ def get_files_to_process(ti):
     """Get files from S3 produced by the discovery task.
     Used as part of both the parallel_run_process_rasters and parallel_run_process_vectors tasks.
     """
-    payload = ti.xcom_pull(task_ids=f"{group_kwgs['group_id']}.discover_from_s3")
-    payloads_xcom = payload.pop("payload", [])
+    dynamic_group_id = ti.task_id.split(".")[0]
```

Comment:
The group ID gets assigned and overwritten automatically when the subdag is used with an `.expand()` or `.chain()` call. This change should allow the task to `xcom_pull` from the correct task which precedes it.